### ISSUE TITLE
False positive `Argument type partially is unknown` within multiple loops.

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4961]
### Discussed in https://github.com/microsoft/pyright/discussions/4959

<div type='discussions-op-text'>

<sup>Originally posted by **Andy-Dihong-Luo** April 15, 2023</sup>
Simplified code example: 
```python
a_value: int = any
a_list: list[int] = any

for _ in range(any):
    for _ in range(any):
        a_value = a_list[a_value]  # Argument type is partially unknown
```

It seems like a bug, which also happens in for-while loop, while-while loop, etc. 
If not, hopefully someone could help solving this. Maybe it's about possible IndexError, but not sure. </div>

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.304, which is now published. It will also be included in a future release of pylance.
